rotkehlchen/rotkehlchen.py

```diff
@@ -481,7 +481,7 @@ def query_fiat_balances(self) -> Dict[Asset, Dict[str, FVal]]:
 
     def query_balances(
             self,
-            requested_save_data: bool = True,
+            requested_save_data: bool = None,
```

Comment:
I am really surprised that the `mypy` tool did not complain for this. Here you are assigning a `None` value to a variable you have specified as a boolean. That should not be the case and that's an error.

Later down you are comparing: 

`if requested_save_data is not False` which is always going to be false by providing a `requested_save_data` with `None` value because if you open a python interpreter you can see `None is False == False`.

Please change this to `False` by default.

Reply:
The reason I did it this way is because if the default is "False" then this `query_balances` will always save if the balance save frequency has lapsed. Since I saw that you have a test which explicitly expects this call to not save if save_data is set to False (even if the frequency has lapsed), I did it like this. From what I saw the interpreter shows:
![image](https://user-images.githubusercontent.com/27592/82881548-1e1a8800-9f40-11ea-85e1-3916c7faab7c.png) therefore this is not always false.

I agree it's not clean, but the only other way if the default boolean value is False that I could think of would be to introduce a new parameter to "force" to indicate whether it should force save or force not save.

Reply:
> I did it like this. From what I saw the interpreter shows:

 This is what you are doing here.

As you see. `requested_save_data is False` returns `False` for `None`. Which is counterintuitive as `None` is consider a Falsy value in python. This is why you should never compare `None` to a boolean in python. You need to be more explicit.

> will always save if the balance save frequency has lapsed

That's fine and that should be the new intended behavior.

> Since I saw that you have a test which explicitly expects this call to not save if save_data is set to False (even if the frequency has lapsed), I did it like this

I think that this needs to change now that we use this variable to force saving.